src/main/java/core/basesyntax/Application.java

```diff
@@ -0,0 +1,22 @@
+package core.basesyntax;
+
+public class Application {
+    private static final int arrayLenght = 6;
+    private static FigureSupplier figureSupplier = new FigureSupplier();
+
+    public static void main(String[] args) {
+        Figure[] figureArray = new Figure[arrayLenght];
+
+        for (int i = 0; i < arrayLenght / 2; i++) {
+            figureArray[i] = figureSupplier.getRandomFigure();
+        }
+
+        for (int i = arrayLenght / 2; i < arrayLenght; i++) {
+            figureArray[i] = figureSupplier.getDefaultFigure();
```

Comment:
You're using the division of 'arrayLenght' multiple times which could lead to performance issues if the array was very large. Consider calculating this value once and storing it in a local variable before the loop.